lib/snip.c

```diff
@@ -449,10 +463,16 @@ grn_snip_add_cond(grn_ctx *ctx, grn_obj *snip,
   grn_snip *snip_;
 
   snip_ = (grn_snip *)snip;
-  if (!snip_ || !keyword || !keyword_len || snip_->cond_len >= MAX_SNIP_COND_COUNT) {
+  if (!snip_ || !keyword || !keyword_len) {
     return GRN_INVALID_ARGUMENT;
   }
 
+  if (snip_->cond_len >= snip_->cond_capacity) {
+    if ((rc = grn_snip_expand_cond(ctx, snip_))) {
```

Comment:
Could you avoid assigning in `if`? I don't like this style except `while ((x = xxx_next()))` case because it causes long `if (...)` code. It's difficult to read.

```suggestion
    rc = grn_snip_expand_cond(ctx, snip_);
    if (rc) {
```